doc/esmvalcore/preprocessor.rst

```diff
@@ -138,344 +260,559 @@ losing generality. To mask ice out, `mask_landseaice` can be used:
       mask_landseaice:
         mask_out: ice
 
-and requires only one argument:
-* mask_out: either landsea or ice.
+and requires only one argument: ``mask_out``: either ``landsea`` or ``ice``.
 
-As in the case of `mask_landsea`, the preprocessor automatically retrieves the
-`fx: sftgif` mask.
+As in the case of ``mask_landsea``, the preprocessor automatically retrieves
+the ``fx_files: [sftgif]`` mask.
 
-Another option is to just read the fx masks as any other CMOR variable and use
-it within a diagnostic script. This can be done in the variable dictionary by
-specifiying the desired fx variables (masks):
+See also :func:`esmvalcore.preprocessor.mask_landseaice`.
 
-.. warning::
-  Code snippet, text and link to function to be added (after #1037 and #1075
-  are closed).
+Mask files
+----------
+
+At the core of the land/sea/ice masking in the preprocessor are the mask files
+(whether it be fx type or Natural Earth type of files); these files (bar
+Natural Earth) can be retrived and used in the diagnostic phase as well or
+solely. By specifying the ``fx_files:`` key in the variable in diagnostic in
+the recipe, and populating it with a list of desired files e.g.:
+
+.. code-block:: yaml
+
+    variables:
+      ta:
+        preprocessor: my_masking_preprocessor
+          fx_files: [sftlf, sftof, sftgif, areacello, areacella]
+
+Such a recipe will automatically retrieve all the ``fx_files: [sftlf, sftof,
+sftgif, areacello, areacella]``-type fx files for each of the variables that
+are needed for and then, in the diagnostic phase, these mask files will be
+available for the developer to use them as they need to. The `fx_files`
+attribute of the big `variable` nested dictionary that gets passed to the
+diagnostic is, in turn, a dictionary on its own, and members of it can be
+accessed in the diagnostic through a simple loop over the ``config`` diagnostic
+variable items e.g.: 
 
+.. code-block::
+
+    for filename, attributes in config['input_data'].items():
+        sftlf_file = attributes['fx_files']['sftlf']
+        areacello_file = attributes['fx_files']['areacello']
+
+.. _masking of missing values:
+
+Missing values masks
+--------------------
+
+Missing (masked) values can be a nuisance especially when dealing with
+multimodel ensembles and having to compute multimodel statistics; different
+numbers of missing data from dataset to datest may introduce biases and
```

Comment:
Should be "dataset" instead of "datest"?

Reply:
Yes.